Part 1: требования, dataset, план алгоритма

evgenya2000 opened the pull request:
Добавлены требования, dataset, план алгоритма

evgenya2000 committed:
Create dataset.md

evgenya2000 committed:
Add files via upload

evgenya2000 committed:
Add files via upload

evgenya2000 committed:
Delete badge.jpg

evgenya2000 committed:
Add files via upload

evgenya2000 committed:
Update dataset.md

evgenya2000 committed:
Update README.md

evgenya2000 committed:
Update README.md

evgenya2000 committed:
Update README.md

evgenya2000 committed:
Update dataset.md

evgenya2000 committed:
Исправления readme 1

Исправила в постановке условия:
    1. Можно вращать предметы
    2. Не камера, а изображение дб не меньше 12 МП
    3.Изменила требования к входным данным (Изображение не обязательно дб горизонтальным, но лист с многоугольником должен находиться слева, а предметы  справа)

evgenya2000 committed:
2 исправления

1. На изображении предмет может быть в любом положении, где плоскость предмета с наибольшей площадью находится параллельно плоскости стола, но при обработке изображения при параллельном переносе предмет вращаться не будет. В датасете есть 4 пример предельного случая, когда ручка может поместиться в прямоугольник при параллельном переносе. Но если бы она была расположена, например, параллельно большой стороны листа, то алгоритм должен был бы вернуть NO
2. Изображение 4272*2848 это разрешение в 12 Мп и это действительно много, исправила на 1 Мп.

evgenya2000 committed:
План алгоритма 1

Добавлен план алгоритма

evgenya2000 committed:
Изменен формат

semen200251 commented on README.md:
```diff
@@ -1 +1,68 @@
+
+  На вход поступает фотография, на которой имеется некоторое множество предметов, ранее выбранных и зафиксированных. Также на фотографии расположен белый лист бумаги А4, на котором черным маркером нарисован многоугольник произвольного размера.
+
+  Алгоритм должен определить, поместятся ли предметы в этот многоугольник с помощью параллельного переноса. Объекты могут поместиться в многоугольнике, если при его наложении на совокупность предметов, никакая из сторон объектов не будет выходить за границы фигуры.
```
Возможно следует уточнить, что предметы не могут не только выходить за границы фигуры, но и лежать на границе.

evgenya2000 replied:
По моей постановке предметы могут лежать на границе, но не могут пересекать её. Это уточнила в своей постановке

semen200251 commented on README.md:
```diff
@@ -1 +1,68 @@
+
+#### Поиск многоугольника и предметов на входном изображении
+     1 Загрузить изображения
+     2 Сжать изображения для ускорения последующих операций
```
Для ускорения, возможно, следовало бы ещё и делать изображение чёрно-белым.

evgenya2000 replied:
Согласна, добавила

evgenya2000 committed:
Изменения по комментариям Хомец Сёмы

1. Предметы могут лежать на границы
2. Для ускорения сделать изображение черно-белым

PIDAMI commented on README.md:
```diff
@@ -1 +1,68 @@
+      Число вершин нарисованного многоугольника не больше 10
+      Края листа бумаги на фото хорошо видны
+#### Выход:
+    В качестве ответа на задачу, в консоль будет выведен идин из ответов:
```
Было бы неплохо, если алгоритм еще выводил картинку с размещением предметов в случае успеха

evgenya2000 replied:
Согласна, добавила

PIDAMI commented on README.md:
```diff
@@ -1 +1,68 @@
+#### Сопоставление многоугольника и предметов
+
+     1 Найти сумму площадей найденных предметов и сравнить с площадью многоугольника, если сумма площадей предметов превосходит площадь многоугольника, то прервать работу алгоритма и вернуть NO
+     2 Параллельным переносом сдвигаем по одному предмету внутрь маски многоугольника.  Проверяем удачно ли размещен предмет внутри маски прямоугольника с использованием побитовой операции AND, если да, то предмет закрепляется а этом месте, если нет, то возвращаем NO
```
Получается, что мы максимум по 1 разу проверим каждый предмет на вместимость в многоугольник? Может стоит как-то попробовать подвигать предметы в поиске возможных размещений вместо того, чтобы сразу заканчивать работу алгоритма при первой неудачной попытке размещения очередного предмета

evgenya2000 replied:
В постановке задачи сказано: "Объекты могут поместиться в многоугольнике, если при его наложении на совокупность предметов, никакая из сторон объектов не будет выходить за границы фигуры (сторона объекта может лежать на границе)" . Поэтому алгоритм и проверяет вместимость только один раз параллельным переносом.

evgenya2000 replied:
Поскольку в постановке задачи говорится о наложении многоугольника на совокупность объектов, то в этом случае алгоритм не должен вращать предметы или искать другие места для размещений

PIDAMI reviewed (commented):
Что понравилось:
+ Идея про сжатие изображений для ускорения вычислений (хотя сходу неясно, как это скажется на корректности алгоритма)

Что не очень понравилось:
+ Слишком упрощенный алгоритм - рассматриваются только параллелльные переносы без возможности как-то перещать уместившиеся объекты внутри многоугольника в случае, когда не получается уместить очередной предмет
+ Небольшое замечание про действия алгоритма при успешном завершении

evgenya2000 committed:
Изменения по комментариям Попова Ивана

Вывод результата подстановки предмета в случае успеха

evgenya2000 committed:
Update README.md

evgenya2000 committed:
Update README.md
